[jscoq] Wait to send coqInfo message until all libinfo has been sent.

ejgallego opened the pull request:
This fixes a race that was made more apparent when the package loading
was moved to Javascript: we tried to load packages before the package
manager knew about them [coqinfo and pkginfo ran in parallel, the
first was executed and the callback triggered loading, getting a
"package not known exception"]

This fixes the Iris example, which now loads; we also remove
whitespace and adjust dependencies so the full mathcomp doesn't have
to be loaded for the start example page.

corwin-of-amber commented on ui-js/coq-manager.js:
```diff
@@ -623,10 +628,10 @@ class CoqManager {
-        
+
         this.packages.waitFor(pkgs)
-        .then(() => this.packages.loadDeps(pkgs))
```
I believe having `.then` aligned to line beginnings is somewhat of an accepted habit in JS. But I will not be picky.

ejgallego replied:
I just followed the automation of my editor here, please feel free to suggest a set of indent rules and I will teach `js-mode` to follow them.

corwin-of-amber commented:
There is a small conflict because of my accidental Node-related commit which is easy to resolve.
Any idea why Travis fails?

ejgallego commented:
> There is a small conflict because of my accidental Node-related commit which is easy to resolve.
> Any idea why Travis fails?

Thanks @corwin-of-amber , indeed somehow I pushed the wrong branch; I've rebased w.r.t. the main branch and removed the indentation changes on `.then`.

ejgallego commented:
My editor is set up to remove trailing whitespace, it may be a bit annoying indeed, so I can remove it if needed. [The thing is that in Coq we have a "linter" that forbids trailing whitespace so I tend to use that stuff]
